cmd/notation/sign.go

```diff
@@ -52,6 +52,9 @@ Example - Sign an OCI artifact using the default signing key, with the default J
 Example - Sign an OCI artifact using the default signing key, with the COSE envelope:
   notation sign --signature-format cose <registry>/<repository>@<digest> 
 
+Example - Sign an OCI artifact with a specified plugin and on-demand remote key
+  notation sign --plugin <plugin_name> --id <remote_key_id> <registry>/<repository>@<digest>
+
```

Comment:
Should this go above the Prerequisite since it doesn't require a signing key to be configured first?

Reply:
@kody-kimberl Good catch. I removed the prerequisite from the example section. It makes the examples look neat. We can explain this prerequisite in Notation documentation.